### PR TITLE
Support decord video decoder

### PR DESCRIPTION
**Pull request by haithamkhedr**
Summary: Support usage of decord backend in video decoding
Test: Tested against torchvision decoding for sanity checks. Both return the same clips for a given random seed

### CODE REVIEW

File: slowfast/datasets/video_container.py [hunk 1]
```diff
@@ -25,5 +30,16 @@ def get_video_container(path_to_vid, multi_thread_decode=False, backend="pyav"):
             # Enable multiple threads for decoding.
             container.streams.video[0].thread_type = "AUTO"
         return container
+    elif backend == "decord":
+        worker_id = torch.utils.data.get_worker_info().id
+        if device == 'cpu':
+            NUM_CPUS = os.cpu_count()
```
Comment (sdebnathusc) [inline slowfast/datasets/video_container.py]: Any specific reason why the variable name is in CAPS?

File: slowfast/datasets/video_container.py [hunk 2]
```diff
@@ -25,5 +30,16 @@ def get_video_container(path_to_vid, multi_thread_decode=False, backend="pyav"):
...
+            ctx = de.cpu(worker_id % NUM_CPUS)
+        else:
+            #NOTE: Install decord from source to support GPU acceleration. Requires some apt pkgs !
+            MAX_GPUS_PER_NODE = torch.cuda.device_count()
```
Comment (sdebnathusc) [inline slowfast/datasets/video_container.py]: Any specific reason why the variable name is in CAPS?

Comment (haithamkhedr) [reply slowfast/datasets/video_container.py]: Those variables were previously hardcoded constants. Resolved

**Review by sdebnathusc — APPROVED** [review]
